### ISSUE TITLE
JSONL requirement in `Predict` is too restrictive

### ISSUE THREAD

**Issue by joelgrus** [allenai/allennlp#653]
for context, I am playing around with using AllenNLP for 

https://www.kaggle.com/c/jigsaw-toxic-comment-classification-challenge

I have trained a model, and now I need to create a CSV of predictions on the test set. The input file is a CSV {"comment_id", "comment_text"}. Right now in order to run `predict` on this, I have to convert it to a JSONL file. That's not hard, but it seems like an unnecessary step.

The relevant code in `predict.py` is

```
    batch_json_data = []
    for line in input_file:
        if not line.isspace():
            # Collect batch size amount of data.
            json_data = json.loads(line)
            batch_json_data.append(json_data)
            if len(batch_json_data) == batch_size:
                _run_predictor(batch_json_data)
                batch_json_data = []
```

Of course we don't want to break backward compatibility. One immediate idea is to give the base `Predictor` class a method

```
def line_to_json(line: str) -> JsonDict:
    return json.loads(line)
```

and then call that (rather than `json.loads`) from `predict.py`. Then in my example I could override it like

```
def line_to_json(line: str) -> JsonDict:
    id, text = parse_csv_line(line)
    return {"id": id, "text": text}
```

and be able to use my `test.csv` rather than have to make a JSONL file.

and then probably we'd do the same for the output:

```
def json_output_to_line(line: str) -> JsonDict:
    return json.dumps(line)
```

and I'd override that too

**Comment by matt-gardner** [comment]
Seems totally reasonable.

**Comment by schmmd** [comment]
Love it!

**Comment by matt-gardner** [comment]
Fixed by #655.
